targets/TARGET_WIZNET/TARGET_W7500x/analogin_api.c

```diff
@@ -57,12 +55,8 @@ void analogin_init(analogin_t *obj, PinName pin)
     // Save pin number for the read function
     obj->pin = pin;
 
-    // The ADC initialization is done once
-    if (adc_inited == 0) {
-        adc_inited = 1;
-
-        ADC_Init();
-    }
+    // The ADC initialization
+    ADC_Init();
```

Comment:
Interesting, hw reset peripheral for each ADC pin. If this is their hw requirement to restart adc like that for each adc channel initialization, ok.